server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -394,16 +384,56 @@ public void write(OutputStream outputStream) throws IOException, WebApplicationE
          .addData("peer", req.getRemoteAddr())
          .emit();
 
+      return context.gotError(e);
+    }
+    finally {
+      Thread.currentThread().setName(currThreadName);
+    }
+  }
+
+  protected class RequestContext
```

Comment:
tried but there were some missing libraries in common module. changed to static class, anyway.